### PR TITLE
Shortest path and all paths

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

**Comment by CPWstatic** [comment]
Jenkins go!

**Comment by CPWstatic** [comment]
Jenkins go!

**Comment by monadbobo** [comment]
Maybe we should support the limit on search depth.  @dutor @sherman-the-tank @dangleptr

File: src/graph/FindPathExecutor.cpp [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+            pd.name = prop;
+            pd.id.set_edge_type(e);
+            props.emplace_back(std::move(pd));
+        }
+    }
+
+    return props;
+}
+
+std::string FindPathExecutor::buildPathString(Path &path) {
```
Comment (laura-ding) [inline src/graph/FindPathExecutor.cpp]: const Path &path

File: src/graph/FindPathExecutor.cpp [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+        auto type = std::get<1>(*step);
+        auto ranking = std::get<2>(*step);
+
+        pathStr += folly::stringPrintf("<%d,%ld>%ld", -type, ranking, id);
+    }
+
+    return pathStr;
+}
+
+cpp2::RowValue FindPathExecutor::buildPathRow(Path &path) {
```
Comment (laura-ding) [inline src/graph/FindPathExecutor.cpp]: ditto

File: src/parser/scanner.lex [hunk 3]
```diff
@@ -122,6 +122,8 @@ LEADER                      ([Ll][Ee][Aa][Dd][Ee][Rr])
 UUID                        ([Uu][Uu][Ii][Dd])
 OF                          ([Oo][Ff])
 DATA                        ([Dd][Aa][Tt][Aa])
+SHORTEST					([Ss][Hh][Oo][Rr][Tt][Ee][Ss][Tt])
```
Comment (laura-ding) [inline src/parser/scanner.lex]: alignment

**Pull request merged**